### PR TITLE
Acl changes for EAP

### PR DESCRIPTION
**Pull request by anurag6**

### CODE REVIEW

File: daq/topology.py [hunk 1]
```diff
@@ -426,6 +426,19 @@ def _generate_main_acls(self):
         local_acl = []
         acls = {}
 
+        dot1x_pri_ports = []
+
+        for devices in self._set_devices.values():
+            for device in devices:
+                dot1x_pri_ports.extend(device.gateway.get_possible_test_ports())
+
+        dot1x_pri_rule = self._ethertype_to_port_acl(0x888e, dot1x_pri_ports)
```
Comment (grafnu) [inline daq/topology.py]: create constant for 0x888e

File: daq/topology.py [hunk 2]
```diff
@@ -531,9 +550,10 @@ def _generate_port_acl(self, port):
         if target_mac:
             assert self._append_acl_template(rules, 'baseline'), 'Missing ACL template baseline'
             self._append_device_default_allow(rules, target_mac)
-            self._write_port_acl(port, rules, file_path)
         else:
-            self._write_port_acl(port, self._make_default_acl_rules(), file_path)
+            rules = self._make_default_acl_rules()
```
Comment (grafnu) [inline daq/topology.py:554]: don't assign directly to rules -- append?

File: daq/topology.py [hunk 3]
```diff
@@ -630,6 +650,13 @@ def _append_device_default_allow(self, rules, target_mac):
             acl = {'rule': subrule}
             self._append_augmented_rule(rules, acl)
 
+    def _ethertype_to_port_acl(self, ether_type, ports):
```
Comment (grafnu) [inline daq/topology.py:653]: can you add eth_type to _make_acl_rule and use that instead?   This function is essentially doing the same work that you're doing here... ?

File: daq/topology.py [hunk 4]
```diff
@@ -519,6 +534,10 @@ def _generate_port_acl(self, port):
         target_mac = None
         rules = []
 
+        if port != self.sec_port:
```
Comment (grafnu) [inline daq/topology.py:537]: why is this check necessary?

File: daq/topology.py (hunk 2 above)
Comment (anurag6) [reply daq/topology.py:554]: I just didn't want to change the functionality of what it was doing. I assume we send the appended rules only for the 'if' case.

File: daq/topology.py (hunk 4 above)
Comment (anurag6) [reply daq/topology.py:537]: Since the ACL is applied to the sec_port, no point trying to add a rule which diverts it back.

Comment (grafnu) [reply daq/topology.py:537]: But this function should never be called with port == self.sec_port
If it is, then that's the core issue.  If you're not sure, then add an assert to verify.

File: daq/topology.py (hunk 2 above)
Comment (grafnu) [reply daq/topology.py:556]: In this case I'd assume it was an oversight in the original code, and just append the rules (keeps the overall pattern of the function more consistent).  If it's a problem (will fail some test somewhere), then we can re-visit.

File: daq/topology.py (hunk 1 above)
Comment (anurag6) [reply daq/topology.py]: Changed.

File: daq/topology.py (hunk 2 above)
Comment (anurag6) [reply daq/topology.py]: Changed.

File: daq/topology.py (hunk 3 above)
Comment (anurag6) [reply daq/topology.py]: Changed.

File: daq/topology.py (hunk 4 above)
Comment (anurag6) [reply daq/topology.py]: Changed.

File: daq/topology.py [hunk 5]
```diff
@@ -426,18 +427,16 @@ def _generate_main_acls(self):
...
+                dot1x_ports.extend(device.gateway.get_possible_test_ports())
 
-        dot1x_pri_rule = self._ethertype_to_port_acl(0x888e, dot1x_pri_ports)
-        incoming_acl.append(dot1x_pri_rule)
+        self._add_acl_rule(incoming_acl, eth_type=self._DOT1X_ETH_TYPE, ports=dot1x_ports)
 
-        dot1x_sec_ports = list(range(1, self.sec_port))
-        dot1x_sec_rule = self._ethertype_to_port_acl(0x888e, dot1x_sec_ports)
-        secondary_acl.append(dot1x_sec_rule)
+        dot1x_ports = list(range(1, self.sec_port))
```
Comment (grafnu) [inline daq/topology.py:438]: don't reuse variable name

Comment (anurag6) [reply daq/topology.py]: Changed.

**Review by grafnu — APPROVED** [review]

**Pull request merged**